docs/spire.md

```diff
@@ -0,0 +1,285 @@
+```
+
+However, the verification materials are removed from the final results as part of the TaskRun status. It is stored in the termination messages (more details below):
+
+```
+$ tkn tr describe cache-image-pipelinerun-8dq9c-fetch-from-git
+...
+<truncated>
+...
+📝 Results
+ NAME                    VALUE
+ ∙ commit       aa79de59c4bae24e32f15fda467d02ae9cd94b01
+ ∙ url          https://github.com/buildpacks/samples
+```
+
+## Architecture Overview
+
+This feature relies on a SPIRE installation. This is how it integrates into the architecture of Tekton:
+
+```
+┌─────────────┐  Register TaskRun Workload Identity           ┌──────────┐
+│             ├──────────────────────────────────────────────►│          │
+│  Tekton     │                                               │  SPIRE   │
+│  Controller │◄───────────┐                                  │  Server  │
+│             │            │ Listen on TaskRun                │          │
+└────────────┬┘            │                                  └──────────┘
+ ▲           │     ┌───────┴───────────────────────────────┐     ▲
+ │           │     │           Tekton TaskRun              │     │
+ │           │     └───────────────────────────────────────┘     │
+ │  Configure│                                          ▲        │ Attest
+ │  Pod &    │                                          │        │   +
+ │  check    │                                          │        │ Request
+ │  ready    │     ┌───────────┐                        │        │ SVIDs
+ │           └────►│  TaskRun  ├────────────────────────┘        │
+ │                 │  Pod      │                                 │
+ │                 └───────────┘     TaskRun Entrypointer        │
+ │                   ▲               Sign Result and update      │
+ │ Get               │ Get SVID      TaskRun status with         │
+ │ SPIRE             │               signature + cert            │
+ │ server            │                                           │
+ │ Credentials       │                                           ▼
+┌┴───────────────────┴─────────────────────────────────────────────────────┐
+│                                                                          │
+│   SPIRE Agent    ( Runs as   )                                           │
+│   + CSI Driver   ( Daemonset )                                           │
+│                                                                          │
+└──────────────────────────────────────────────────────────────────────────┘
+```
+
+Initial Setup:
+1. As part of the SPIRE deployment, the SPIRE server attests the agents running on each node in the cluster.
+2. The Tekton Controller is configured to have workload identity entry creation permissions to the SPIRE server.
+3. As part of the Tekton Controller operations, the Tekton Controller will retrieve an identity that it can use to talk to the SPIRE server to register TaskRun workloads.
```

Comment:
NIT: using 1. 2. 3. here and then 1. 1. 1. below - perhaps it should be 1. 1. 1. here too

Reply:
Done.